src/main/java/teammates/logic/core/AccountsLogic.java

```diff
@@ -238,4 +242,42 @@ private void createStudentAccount(StudentAttributes student)
         accountsDb.createEntity(account);
     }
 
+    /**
+     * Updates the read notifications of an account.
+     */
+    public AccountAttributes updateReadNotifications(String googleId, String notificationId, Instant endTime)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        AccountAttributes a = accountsDb.getAccount(googleId);
+        if (a == null) {
+            throw new EntityDoesNotExistException("Invalid account id.");
+        }
+
+        if (endTime.isBefore(Instant.now())) {
+            throw new InvalidParametersException("Notification has expired.");
+        }
+        Map<String, Instant> updatedReadNotifications = a.getReadNotifications();
+
+        for (Map.Entry<String, Instant> notification : updatedReadNotifications.entrySet()) {
+            if (notification.getValue().isBefore(Instant.now())) {
+                updatedReadNotifications.remove(notification.getKey());
+            }
+        }
```

Comment:
Could add a comment here, might not be obvious as to why the expired notifications are removed